### PR TITLE
En 880 side widget component

### PR DESCRIPTION
**Pull request by fedelocci**

### CODE REVIEW

**Comment by entando-service** [comment]
test this please

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: sass/pages/config/ToolbarPageConfig.scss [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    z-index: 10;
+    overflow-y: auto;
+  }
+
+  &__drawer-pf-sidebar-right > a:hover &__right-bar-title > i  {
+    color: $color-pf-blue-300;
+    text-decoration: none;
+  }
+
+  &__drawer-pf-sidebar-right > a:focus {
```
Comment (alessioscalici) [inline sass/pages/config/ToolbarPageConfig.scss]: Here too

File: src/locales/it.js [hunk 2]
```diff
@@ -25,14 +25,15 @@ export default {
     'app.code': 'Codice',
     'app.used': 'usato',
     'app.actions': 'Azioni',
+    'menu.dashboard': 'Dashboard',
```
Comment (alessioscalici) [inline src/locales/it.js]: Put this back with menu keys

File: src/ui/pages/config/ContentWidgetElement.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          fa-puzzle-piece
+          list-view-pf-icon-sm
+          ContentWidgetElement__widget-icon
+          fa-banner-main-left"
+        />
+      </div>
+      <div className="list-view-pf-body">
+        <div className="list-view-pf-description">
+          <div className="ContentWidgetElement__description list-group-item-heading">
+            <a href="#" title={description}>
```
Comment (alessioscalici) [inline src/ui/pages/config/ContentWidgetElement.js]: Please remove `#`

File: src/ui/pages/config/ContentWidgetElement.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+          fa fa-default
+          fa-puzzle-piece
+          list-view-pf-icon-sm
+          ContentWidgetElement__widget-icon
+          fa-banner-main-left"
+        />
+      </div>
+      <div className="list-view-pf-body">
+        <div className="list-view-pf-description">
+          <div className="ContentWidgetElement__description list-group-item-heading">
```
Comment (alessioscalici) [inline src/ui/pages/config/ContentWidgetElement.js]: Please remove `list-group-item-heading`

File: src/state/page-config/reducer.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { combineReducers } from 'redux';
+import { SET_CONTENT_TOOLBAR, SET_SEARCH_FILTER, CHANGE_VIEW_LIST } from 'state/page-config/types';
+
+export const content = (state = 'WIDGET_LIST', action = {}) => {
```
Comment (alessioscalici) [inline src/state/page-config/reducer.js]: Create constant WIDGET_LIST

File: src/ui/pages/config/ToolbarPageConfig.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+              <FormattedMessage id="app.pages" />
+            </span>
+            <span className="ToolbarPageConfig__open-button-menu-right pull-right">
+              <i className="fa fa-angle-right" aria-hidden="true" />
+            </span>
+          </span>
+        </a>
+        <div className="panel-group">
+          <div className="ToolbarPageConfig__drawer-pf-container" >
+            {this.props.content === 'WIDGET_LIST' ? <ContentWidgetContainer /> : null }
```
Comment (alessioscalici) [inline src/ui/pages/config/ToolbarPageConfig.js]: Create constant WIDGET_LIST

File: src/ui/pages/config/ToolbarPageConfig.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+
+ToolbarPageConfig.propTypes = {
+  onWillMount: PropTypes.func,
+  content: PropTypes.string,
+};
+
+ToolbarPageConfig.defaultProps = {
+  onWillMount: () => {},
+  content: 'WIDGET_LIST',
```
Comment (alessioscalici) [inline src/ui/pages/config/ToolbarPageConfig.js]: Create constant WIDGET_LIST

File: test/state/page-config/reducer.test.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
+import { setContentToolbar, setSearchFilter, changeViewList } from 'state/page-config/actions';
+import reducer from 'state/page-config/reducer';
+
+const DEFAULT_CONTENT = 'WIDGET_LIST';
```
Comment (alessioscalici) [inline test/state/page-config/reducer.test.js]: Use constant WIDGET_LIST

File: test/state/page-config/reducer.test.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  beforeEach(() => {
+    state = reducer();
+  });
+
+  describe('default state', () => {
+    it('should define a content', () => {
+      expect(state.content).toBeDefined();
+      expect(typeof state.content).toBe('string');
+    });
+    it('should define a searchFilter', () => {
```
Comment (alessioscalici) [inline test/state/page-config/reducer.test.js]: Add a newline between its

File: test/state/page-config/selectors.test.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+));
+
+getListWidget.mockReturnValue(WIDGET_LIST.payload);
+
+describe('state/page-config/selectors', () => {
+  it('getPageConfig(state) return a pageConfig object', () => {
+    const pageConfig = getPageConfig(MOCK_STATE);
+    expect(pageConfig).toMatchObject(MOCK_STATE.pageConfig);
+  });
+  it('verify getSearchFilter selector', () => {
```
Comment (alessioscalici) [inline test/state/page-config/selectors.test.js]: Newlines :)

File: test/ui/pages/config/ContentWidget.test.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+  it('has the PageConfigPage class', () => {
+    expect(component.hasClass('ContentWidget')).toBe(true);
+  });
+
+  describe('props', () => {
+    it('widgetList empty data', () => {
+      component = shallow(<ContentWidget widgetList={WIDGET_LIST_MOCK_EMPTY} />);
+      expect(component.find('.ContentWidgetElement__widget-spacer').exists()).toBeFalsy();
+      expect(component.find('.ContentWidgetElement__widget-spacer')).toHaveLength(0);
+    });
```
Comment (alessioscalici) [inline test/ui/pages/config/ContentWidget.test.js]: Newlines :)

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Pull request merged**